### DISCUSSION TITLE
GDB not stopping at break point when using Makefile Tools extension

### DISCUSSION THREAD

**Discussion by userAman16** [microsoft/vscode-cpptools#7741] [Q&A]
Hi ,

I am trying to debug https://github.com/snap-stanford/snap/tree/master/examples/cesna
i have build the project using [makefile-tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools) with [Makefile](https://github.com/snap-stanford/snap/blob/master/examples/cesna/Makefile) 
after the binary got generated.  I am trying to debug with option "DEBUG" in makefile-tools

I have added the -g flag to C++

But it is not hitting the break point and exiting with debug log

Log from debug console

=thread-group-added,id="i1"
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
Warning: Debuggee TargetArchitecture not detected, assuming x86_64.
=cmd-param-changed,param="pagination",value="off"
Stopped due to shared library event (no libraries added or removed)
Loaded '/lib64/ld-linux-x86-64.so.2'. Symbols loaded.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Breakpoint 1, 0x000055555558bc00 in main ()
Loaded '/lib/x86_64-linux-gnu/libstdc++.so.6'. Symbols loaded.
Loaded '/lib/x86_64-linux-gnu/libm.so.6'. Symbols loaded.
Loaded '/lib/x86_64-linux-gnu/libgomp.so.1'. Symbols loaded.
Loaded '/lib/x86_64-linux-gnu/libgcc_s.so.1'. Symbols loaded.
Loaded '/lib/x86_64-linux-gnu/libpthread.so.0'. Symbols loaded.
Loaded '/lib/x86_64-linux-gnu/libc.so.6'. Symbols loaded.
Loaded '/lib/x86_64-linux-gnu/libdl.so.2'. Symbols loaded.
[New Thread 0x7ffff7a0f700 (LWP 120627)]
[New Thread 0x7ffff720e700 (LWP 120628)]
[New Thread 0x7ffff632a700 (LWP 120629)]
[Thread 0x7ffff632a700 (LWP 120629) exited]
[Thread 0x7ffff720e700 (LWP 120628) exited]
[Thread 0x7ffff7a0f700 (LWP 120627) exited]
[Inferior 1 (process 120622) exited normally]
The program '/home/horizon/repo/snap/snap/examples/cesna/cesna' has exited with code 0 (0x00000000).

**Comment by djairao** [comment]
Are you compiling with optimization turned on via a `-O2` or `-O3` flag (or one of the other optimization flags)? If so then possibly that code has been optimized out and the debugger will not stop at the line. If not, try to compile your code with the `-O0` flag to see if that helps.

Of course, double-check to ensure you are providing the necessary inputs for that line of code to be actually executed.
